mlir/lib/Target/Cpp/TranslateToCpp.cpp

```diff
@@ -771,6 +771,22 @@ static LogicalResult printOperation(CppEmitter &emitter, ModuleOp moduleOp) {
   return success();
 }
 
+static LogicalResult printFunctionArgs(CppEmitter &emitter,
+                                       Operation *functionOp,
+                                       ArrayRef<Type> arguments) {
+  raw_indented_ostream &os = emitter.ostream();
+
+  if (failed(interleaveCommaWithError(
+          arguments, os, [&](Type arg) -> LogicalResult {
+            if (failed(emitter.emitType(functionOp->getLoc(), arg)))
+              return failure();
+            return success();
+          })))
+    return failure();
+
+  return success();
```

Comment:
Just return the LogicalResult directly instead of
```c++
if (failed(...)) {
  return failure();
}
return success();
```

```suggestion
  return interleaveCommaWithError(
      arguments, os, [&](Type arg) -> LogicalResult {
        return emitter.emitType(functionOp->getLoc(), arg);
      });
```

Reply:
That is a nearly 1:1 takeover from the existing `printFunctionArgs` implementation. Do you intended that I fix this as part of this PR as well?

Reply:
At least the outer if can be dropped there as well. Would make sense to me to keep this is sync.